pythran/pythonic/include/numpy/all.hpp

```diff
@@ -12,13 +12,19 @@ namespace pythonic
   namespace numpy
   {
     template <class E>
-    bool _all(E begin, E end, utils::int_<1>);
+    typename std::enable_if<types::is_numexpr_arg<E>::value, bool>::type
+    all(E const &expr, types::none_type _ = types::none_type());
 
-    template <class E, size_t N>
-    bool _all(E begin, E end, utils::int_<N>);
+    template <class E>
+    typename std::enable_if<
+        std::is_scalar<E>::value or types::is_complex<E>::value, bool>::type
```

Comment:
On Wed, Sep 02, 2015 at 11:24:03AM -0700, pbrunet wrote:
> >  
> > -    template <class E, size_t N>
> > -    bool _all(E begin, E end, utils::int_<N>);
> > +    template <class E>
> > +    typename std::enable_if<
> > +        std::is_scalar<E>::value or types::is_complex<E>::value, bool>::type
> 
> Don't we have a is_number for this?

This function does not exist, but it definitively should! I implemented
it as is_dtype